test/extensions/filters/http/cache/simple_http_cache/simple_http_cache_test.cc

```diff
@@ -372,6 +373,71 @@ TEST_F(SimpleHttpCacheTest, UpdateHeadersDisabledForVaryHeaders) {
   EXPECT_TRUE(expectLookupSuccessWithHeaders(lookup(request_path_1).get(), response_headers_1));
 }
 
+TEST_F(SimpleHttpCacheTest, UpdateHeadersSkipEtagHeader) {
+  const std::string request_path_1("/name");
+  Http::TestResponseHeaderMapImpl response_headers_1{{"date", formatter_.fromTime(current_time_)},
+                                                     {"cache-control", "public,max-age=3600"},
+                                                     {"etag", "0000-0000"}};
+  insert(request_path_1, response_headers_1, "body");
+  EXPECT_TRUE(expectLookupSuccessWithHeaders(lookup(request_path_1).get(), response_headers_1));
+
+  // Update the date field in the headers
+  time_source_.advanceTimeWait(Seconds(3601));
+  std::string time_now = formatter_.fromTime(current_time_);
+  Http::TestResponseHeaderMapImpl response_headers_2{
+      {"date", time_now}, {"cache-control", "public,max-age=3600"}, {"etag", "1111-1111"}};
+  // The etag header should not be updated
+  Http::TestResponseHeaderMapImpl response_headers_3{
+      {"date", time_now}, {"cache-control", "public,max-age=3600"}, {"etag", "0000-0000"}};
+
+  updateHeaders(request_path_1, response_headers_2, {current_time_});
+
+  EXPECT_TRUE(expectLookupSuccessWithHeaders(lookup(request_path_1).get(), response_headers_3));
+}
+
+TEST_F(SimpleHttpCacheTest, UpdateHeadersSkipSpecificHeaders) {
+  const std::string request_path_1("/name");
+
+  // Vary not tested because we have separate tests that cover it
+  Http::TestResponseHeaderMapImpl response_headers_1{
+      {"date", formatter_.fromTime(current_time_)},
+      {"cache-control", "public,max-age=3600"},
+      {"content-range", "bytes 200-1000/67589"},
+      {"content-length", "800"},
+      {"content-type", "text/html; charset=UTF-8"},
+      {"etag", "0000-0000"},
+      {"link", "<https://example.com>; rel=\"preconnect\""}};
+  insert(request_path_1, response_headers_1, "body");
+  EXPECT_TRUE(expectLookupSuccessWithHeaders(lookup(request_path_1).get(), response_headers_1));
+  time_source_.advanceTimeWait(Seconds(100));
+
+  Http::TestResponseHeaderMapImpl response_headers_2{
+      {"date", formatter_.fromTime(current_time_)},
+      {"cache-control", "public,max-age=3600"},
+      {"content-range", "bytes 5-1000/67589"},
+      {"content-length", "995"},
+      {"content-type", "application/json; charset=UTF-8"},
+      {"age", "20"},
+      {"etag", "1111-1111"},
+      {"link", "<https://changed.com>; rel=\"preconnect\""}};
+
+  // The skipped headers should not be updated
+  // Except "age" should be come 0 because update_header updates the metadata as well
+  Http::TestResponseHeaderMapImpl response_headers_3{
+      {"date", formatter_.fromTime(current_time_)},
+      {"cache-control", "public,max-age=3600"},
+      {"content-range", "bytes 200-1000/67589"},
```

Comment:
can you cover a case where the input headers you are processing have a case where a header appears more than once?

Reply:
the case you have below is good; keep it. But I was thinking of a case where one of the headers in your update-list got repeated. Could you add one of those also?

Reply:
In the original entry, I repeated `etag` twice. In the incoming headermap, I added `content-length` twice. Although neither makes sense in a real-world scenario, these two cases should make sure the code is behaving correctly even if there's some issues with the client.